Documentation/Concepts/events.md

```diff
@@ -0,0 +1,133 @@
+
+GUIDs also solve the problem of having duplicate names, it's not hard to imagine having to have multiple events with the type of `CustomerRegisterd` in your code coming from different microservices.
+{{< /alert >}}
+
+#### Generations
+As the code changes, the event type is also bound to change at some point. These iterations on the same event type are called _generations_. Whenever you add or change a property in an event you should increase this number. This way the filters and handlers can handle a specific generation of an event.
+
+For example, imagine an older generation of an event type didn't have a property. You can have a separate handler for that particular generation that handles the missing property.
+```csharp
+// C# like pseudo-code
+[EventType("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 1)]
+DishPrepared {
+    Dish string;
+}
+
+// Generation 2
+[EventType("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 2)]
+DishPrepared {
+    Dish string;
+    Chef string;
+}
+
+// handlers generations separately
+[Handle("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 1)]
+Handle(DishPrepared @event) {
+    // handle the missing property
+}
+
+[Handle("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 2)]
+Handle(DishPrepared @event) { ... }
+```
+
+{{< alert title="Production ready" color="info" >}}
+For making development easier, you shouldn't worry about incrementing the generation until you're in production.
+{{< /alert >}}
+
+### Public vs. Private
+There is a basic distinction between private events and public events. In much the same way that you would not grant access to other applications to your internal database, you do not allow other applications to subscribe to your private events.
+
+Private events are only accessible within a single [Tenant]({{< ref "tenant" >}}) so that an event committed for one tenant cannot be handled outside of that tenant. Private events model the system within its domain and should be named in a way that makes sense inside the domain.
+
+Public events are also accessible within a single tenant but they can also be added to a public [Stream]({{< ref "event-handlers" >}}) for other microservices to consume.
+
+Your external stream of public events is your contract to the outside world, your API.
+
+{{< alert title="Changes to public events" color="primary" >}}
+Extra caution should be paid to changing public events so as not to break other microservices consuming those events.
+{{< /alert >}}
+
+## Commit vs Publish
+We use the word `Commit` rather than `Publish` when talking about saving events to the [Event Store]({{< ref "event_store" >}}). We want to emphasize that it's the event store that is the source of truth in the system. The act of calling filters/event handlers comes _after_ the event has been committed to the event store.
```

Comment:
To add my 0.02$ here I would also talk about how when talking about "Publish" you'd usually think about publishing events to specific streams (or Topics) and event handlers, microservices etc. While that is not the case for us

Reply:
Added some extra sentences explaining this and the "globalness" of committing an event